### PR TITLE
CLIMATE-623 - Allow propogation of 2D arrays

### PR DESCRIPTION
**Pull request by kwhitehall**
- update subset function in DSP to subset 2D (lat,lon) arrays
- update Dataset class to allow 2D (lat,lon) arrays
- update associated test in test_dataset.py to allow 2D arrays

### CODE REVIEW

**Comment by kwhitehall** [comment]
The change in dataset_processor.py is very cumbersome. I couldn't think of a way to write it more succinctly. @cgoodale @MJJoyce @MBoustani  any suggestions would be greatly appreciated.

File: ocw/dataset.py [hunk 1]
```diff
@@ -180,17 +180,23 @@ def _validate_inputs(self, lats, lons, times, values):
...
-        elif values.shape != (time_count, lat_count, lon_count):
+        if value_dim == 2 and values.shape != (lat_count, lon_count):
             err_msg = """Value Array must be of shape (times, lats, lons).
-Expected shape (%s, %s, %s) but received (%s, %s, %s)""" % (time_count,
-                                                            lat_count,
-                                                            lon_count,
-                                                            values.shape[0],
-                                                            values.shape[1],
-                                                            values.shape[2])
+    Expected shape (%s, %s, %s) but received (%s, %s, %s)""" % (lat_count,
```
Comment (cgoodale) [inline ocw/dataset.py]: Need to reduce the string inputs to be 4 instead of 6.

Comment (kwhitehall) [inline ocw/dataset.py]: Thanks @cgoodale! Will update

**Comment by MBoustani** [comment]
Looks good to me @kwhitehall .

**Pull request merged**